### PR TITLE
chore: clean expire data from new schema tables

### PR DESCRIPTION
**Pull request by dengmingtong**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

clear history data including:
event_v2
item_v2
user_v2
session
all SP tables

## Implementation highlights

e2e test result:
If we set ClearExpiredEventsRetentionRangeDays as 20, the latest_timestamp is 2024-04-16,
after clearing data, the min timestamp is 2024-03-27

## Test checklist

- [ ] add new test cases
- [x] all code changes are covered by unit tests
- [x] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [x] deploy data modeling
    - [x] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting
  - [ ] streaming ingestion
    - [ ] with Redshift Serverless
    - [ ] with provisioned Redshift

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: src/analytics/lambdas/custom-resource/create-schemas.ts [hunk 1]
```diff
@@ -265,6 +265,10 @@ function getCreateOrUpdateSchemasSQL(newAddedAppIdList: string[], props: Resourc
       table_event_parameter: odsTableNames.event_parameter,
```
Comment (zxkane) [inline src/analytics/lambdas/custom-resource/create-schemas.ts:265]: Should we remove these legacy tables?

File: src/analytics/lambdas/custom-resource/create-schemas.ts [hunk 2]
```diff
@@ -323,6 +327,10 @@ function getCreateOrUpdateViewForReportingSQL(newAddedAppIdList: string[], props
...
```
Comment (zxkane) [inline src/analytics/lambdas/custom-resource/create-schemas.ts:327]: Should we remove these legacy tables?

File: src/analytics/lambdas/refresh-materialized-views-workflow/check-next-refresh-view.ts [hunk 3]
```diff
@@ -18,6 +18,7 @@ export interface RefreshViewOrSp {
   name: string;
   type: string;
   timezoneSensitive: string;
+  viewName?: string;
```
Comment (zxkane) [inline src/analytics/lambdas/refresh-materialized-views-workflow/check-next-refresh-view.ts:21]: Is it the same as the `name`?

File: src/analytics/private/model.ts [hunk 4]
```diff
@@ -211,6 +211,10 @@ export type MustacheParamType = {
   table_clickstream_log: string;
   table_refresh_mv_sp_status: string;
   table_ods_events: string;
```
Comment (zxkane) [inline src/analytics/private/model.ts:213]: clean the legacy tables

**Review by zxkane — COMMENTED** [review]
- [ ] add e2e test result

**Review by amliuyong — APPROVED** [review]

File: src/analytics/lambdas/custom-resource/create-schemas.ts (hunk 1 above)
Comment (dengmingtong) [reply src/analytics/lambdas/custom-resource/create-schemas.ts:265]: sp-migrate-ods-events-1.0-to-1.1.sql still use it.

File: src/analytics/lambdas/custom-resource/create-schemas.ts (hunk 2 above)
Comment (dengmingtong) [reply src/analytics/lambdas/custom-resource/create-schemas.ts:327]: sp-migrate-ods-events-1.0-to-1.1.sql still use it.

File: src/analytics/lambdas/refresh-materialized-views-workflow/check-next-refresh-view.ts (hunk 3 above)
Comment (dengmingtong) [reply src/analytics/lambdas/refresh-materialized-views-workflow/check-next-refresh-view.ts:21]: the name is sp name, viewName is the table name that is created after sp refreshed
for example:
name: clickstream_acquisition_country_new_user_sp
viewName: clickstream_acquisition_country_new_user

File: src/analytics/private/model.ts (hunk 4 above)
Comment (dengmingtong) [reply src/analytics/private/model.ts:213]: sp-migrate-ods-events-1.0-to-1.1.sql still use it.

File: src/analytics/lambdas/custom-resource/create-schemas.ts (hunk 2 above)
Comment (zxkane) [reply src/analytics/lambdas/custom-resource/create-schemas.ts:327]: Let's remove the sql too. It does not make sense to keep it in the latest version.

Comment (dengmingtong) [reply src/analytics/lambdas/custom-resource/create-schemas.ts]: refactored it

File: src/analytics/private/model.ts (hunk 4 above)
Comment (dengmingtong) [reply src/analytics/private/model.ts]: refactored it

File: src/analytics/lambdas/custom-resource/create-schemas.ts (hunk 1 above)
Comment (dengmingtong) [reply src/analytics/lambdas/custom-resource/create-schemas.ts]: refactored it

File: src/analytics/lambdas/custom-resource/create-schemas.ts [hunk 5]
```diff
@@ -262,9 +262,12 @@ function getCreateOrUpdateSchemasSQL(newAddedAppIdList: string[], props: Resourc
       database_name: props.projectId,
       schema: app,
       table_event: odsTableNames.event,
-      table_event_parameter: odsTableNames.event_parameter,
```
Comment (zxkane) [inline src/analytics/lambdas/custom-resource/create-schemas.ts:265]: You need to remove all legacy tables, event, user, and item.

**Review by zxkane — CHANGES_REQUESTED** [review]

**Review by zxkane — APPROVED** [review]

File: src/analytics/lambdas/custom-resource/create-schemas.ts (hunk 5 above)
Comment (dengmingtong) [reply src/analytics/lambdas/custom-resource/create-schemas.ts:265]: Remove all

**Pull request merged**